exporter/metric/metric.go

```diff
@@ -374,12 +370,30 @@ func recordToTypedValueAndTimestamp(r *export.Record) (*monitoringpb.TypedValue,
 			StartTime: &googlepb.Timestamp{
 				Seconds: now,
 			},
-		}
-		return &monitoringpb.TypedValue{
-			Value: &monitoringpb.TypedValue_Int64Value{
-				Int64Value: value,
+			EndTime: &googlepb.Timestamp{
+				Seconds: now,
 			},
-		}, t, nil
+		}
+		switch kind {
+		case apimetric.Int64NumberKind:
+			return &monitoringpb.TypedValue{
+				Value: &monitoringpb.TypedValue_Int64Value{
+					Int64Value: value,
+				},
```

Comment:
This code is repeated so may be it could be converted to a func.

Reply:
Worrysome part here is that only Counter returns int64 where as the other kinds return apimetric.Number.

The combination of this and the comment by @nilebox looks better, i.e. extracting each cases to functions and in addition to it make TypeValue part a function.

WDYT?